### DISCUSSION TITLE
unauthorized in api guard

### DISCUSSION THREAD

**Discussion by ahmed2104** [adonisjs/core#1318] [Help] [answered]
when i use middleware auth and put bearer token in insomnia , it gives me unauthorized and i followed all steps of documentation

**Comment by thetutlage** [comment, marked as answer]
What information do you expect us to use here to help you?

**Reply by ahmed2104** [reply]
i don't know , it should works when i follow documentation

**Reply by thetutlage** [reply]
Cool, then there is hardly anything I can help you with :)

**Reply by ahmed2104** [reply]
so , what should i do ?

**Reply by thetutlage** [reply]
Provide the required information that me or someone else can use to see what exactly the problem is.

**Reply by ahmed2104** [reply]
i just followed documentation and i put registermiddleware in kernel.ts and made this route 
Route.get('dashboard' , 'DataController.datadata').middleware('auth')
and it does not work
note i made everything in AuthController like documentation

**Reply by ahmed2104** [reply]
*/
Server.middleware.registerNamed({
  auth: 'App/Middleware/Auth',
})

**Reply by ahmed2104** [reply]
public async login ({ response, request, auth }: HttpContextContract) {
        const email = request.input('email')
        const password = request.input('password')
        const token = await auth.use('api').attempt(email, password)
        return response.json(token)
      }

    public async register ({ response , request , auth }: HttpContextContract){
        const { username , email , password } = request.all()
        const user = new User()
        user.username = username
        user.email = email
        user.password = password
        const data = await user.save()
        const token = await auth.use('api').attempt(email, password)
        return response.json({ data : data , token : token })
    }

**Reply by ahmed2104** [reply]
solution ?

**Reply by ahmed2104** [reply]
i solved my problem by myself , thx
